### PR TITLE
[stdlib] Fix error in nextUp and nextDown

### PR DESCRIPTION
**Pull request by natecook1000**
#### What's in this pull request?

This fixes an error in the `nextUp` and `nextDown` implementation and adds a test for verification.

Before:

``` swift
(swift) (10.nextUp, 10.nextDown)
// r0 : (Double, Double) = (10.000000000000002, -10.000000000000002)
(swift) ((-10).nextUp, (-10).nextDown)
// r1 : (Double, Double) = (10.000000000000002, -10.000000000000002)
```

After:

``` swift
(swift) (10.nextUp, 10.nextDown)
// r0 : (Double, Double) = (10.000000000000002, 9.9999999999999982)
(swift) ((-10).nextUp, (-10).nextDown)
// r1 : (Double, Double) = (-9.9999999999999982, -10.000000000000002)
```
#### Resolved bug number: n/a

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

File: test/1_stdlib/FloatingPoint.swift.gyb [hunk 1]
```diff
@@ -171,6 +171,42 @@ FloatingPoint.test("Double/HashValueZero") {
   expectEqual(zero.hashValue, negativeZero.hashValue)
 }
 
+FloatingPoint.test("Float/NextUp") {
+  let values: [Float] = [-20, -2, -1, -0, 0, 1, 2, 20]
```
Comment (gribozavr) [inline test/1_stdlib/FloatingPoint.swift.gyb]: Could you change the tests to check the output exactly?

```swift
/// Pairs of consecutive floating point numbers as ordered by `nextUp`/`nextDown`.
let nextUpDownTests: [(Float, Float)] = [
  (Float.nan, Float.nan),
  (Float.infinity, Float.infinity),
  (-Float.infinity, -Float.infinity),
  (0x1.fffffep-1, 1.0), (1.0, 0x1.000002p+0),
  (-0x1p-149, 0.0), (0.0, 0x1p-149),
  (0x1.effffep-1, 0.96875), (0.96875, 0x1.f00002p-1),
]

FloatingPoint.test("Float.nextUp, .nextDown")
  .forEach(in: nextUpDownTests) {
  (prev, succ) in
  expectEqual(succ.bitPattern, prev.nextUp.bitPattern)
  expectEqual(prev.bitPattern, succ.nextDown.bitPattern)
  expectEqual((-succ).bitPattern, (-prev).nextDown.bitPattern)
  expectEqual((-prev).bitPattern, (-succ).nextUp.bitPattern)  
}
```

File: stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb [hunk 2]
```diff
@@ -238,25 +238,25 @@ public func expectOptionalEqual${Generic}(
...
 
-public func expectGT(_ lhs: Int, _ rhs: Int, ${TRACE}) {
+public func expectGT<T: Comparable>(_ lhs: T, _ rhs: T, ${TRACE}) {
   if !(lhs > rhs) {
     expectationFailure("\(lhs) > \(rhs)", trace: ${trace})
   }
 }
 
-public func expectGE(_ lhs: Int, _ rhs: Int, ${TRACE}) {
+public func expectGE<T: Comparable>(_ lhs: T, _ rhs: T, ${TRACE}) {
```
Comment (gribozavr) [inline stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb]: In the standard library we usually add a space before the colon as well.

**Comment by lattner** [comment]
@stephentyrone just fyi

File: test/1_stdlib/FloatingPoint.swift.gyb (hunk 1 above)
Comment (natecook1000) [inline test/1_stdlib/FloatingPoint.swift.gyb]: Thanks, I didn't know about that `forEach` method! I've updated the tests.

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
Thank you!

**Pull request merged**